[Badge][base] Drop `component` prop (#37028)
mui/material-ui: docs/data/base/components/badge/badge.md

</span>
```

### Custom structure

> Use the `slots.root` prop to override the root slot with a custom element:

Reviewer: I'd remove this paragraph and change the next one to: "Use the `slots` prop to override the root or any other interior slot."

<Badge slotProps={{ badge: { className: 'my-badge' } }} />
```

#### Usage with TypeScript

> In TypeScript, you can specify the custom component type used in the `slots.root` as a generic to the unstyled component. This way, you can safely provide the custom compoenent's props directly on the compnent:

Reviewer: ```suggestion
In TypeScript, you can specify the custom component type used in the `slots.root` as a generic parameter of the unstyled component. This way, you can safely provide the custom root's props directly on the component:
```

Author: will do the same for other components' markdown too
